### ISSUE TITLE
evm_revert occasionally crashes testrpc

### ISSUE THREAD

**Issue by shrugs** [ConsenSys-archive/ganache-cli-archive#426]
My testrpc instance occasionally crashes directly after an `evm_revert` rpc call is issued, complaining about `toJSON`. Relevant output:

```
evm_revert
   > {
   >   "jsonrpc": "2.0",
   >   "method": "evm_revert",
   >   "id": 1510815434392,
   >   "params": [
   >     "0x01"
   >   ],
   >   "external": true
   > }
Reverting to snapshot #1
/Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:71492
var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=false;var ENVIRONMENT_IS_WORKER=false;var ENVIRONMENT_IS_NODE=false;var ENVIRONMENT_IS_SHELL=false;if(Module["ENVIRONMENT"]){if(Module["ENVIRONMENT"]==="WEB"){ENVIRONMENT_IS_WEB=true}else if(Module["ENVIRONMENT"]==="WORKER"){ENVIRONMENT_IS_WORKER=true}else if(Module["ENVIRONMENT"]==="NODE"){ENVIRONMENT_IS_NODE=true}else if(Module["ENVIRONMENT"]==="SHELL"){ENVIRONMENT_IS_SHELL=true}else{throw new Error("The provided Module['ENVIRONMENT'] value is not valid. It must be one of: WEB|WORKER|NODE|SHELL.")}}else{ENVIRONMENT_IS_WEB=typeof window==="object";ENVIRONMENT_IS_WORKER=typeof importScripts==="function";ENVIRONMENT_IS_NODE=typeof process==="object"&&"function"==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT

TypeError: Cannot read property 'toJSON' of null
    at /Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:88023:25
    at /Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:25374:5
    at /Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:85943:7
    at /Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:7174:16
    at replenish (/Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:7694:25)
    at iterateeCallback (/Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:7684:17)
    at /Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:7659:16
    at /Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:85938:9
    at Object.decode (/Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:46720:5)
    at /Users/matt/Projects/steak.network/token/node_modules/ethereumjs-testrpc/build/cli.node.js:85935:20
```

## Expected Behavior
Testrpc should revert to snapshot with id `1`.

## Current Behavior
Testrpc cli crashes.

## Steps to Reproduce (for bugs)
Again, this is only "occasionally" (which I'm aware is infuriating), so apologies for the lack of concrete information.

This bug is triggered by truffle's `before` block in tests when it attempts to roll back to a previous snapshot. Removing the truffle-specific snapshotting code seems to fix the issue. It might be a race condition with saving snapshots, given the "occasional" nature of the issue.

I can put some effort into producing a minimal example if necessary.

## Your Environment
- macOS Sierra
- ethereumjs-testrpc@6.0.2
- truffle@4.0.1

**Comment by shrugs** [comment]
this is related to https://github.com/trufflesuite/ganache/issues/82

**Comment by benjamincburns** [comment]
@Shrugs if you have time to get together a more reliable repro/minimal example, I'd be much obliged.

And either way, thanks for reporting!

**Comment by shrugs** [comment]
Haven't had the time to construct a project, but this also occurs occasionally after running the zeppelin-solidity tests:

```
truffle test --network development
```

(which isn't the correct way to run those tests, mind you; `npm run test` is the preferred method)

**Comment by benjamincburns** [comment]
I've just published `ganache-cli@7.0.0-beta.0` which I believe fixes this bug, as well as adds support for websockets, and contains numerous other fixes. You can get it by doing `npm install -g ganache-cli@beta`.

Closing this case for now, but please reopen if this problem shows its ugly face again.
